test/framework/toolchain.py

```diff
@@ -339,6 +353,20 @@ def test_optarch_generic(self):
                         else:
                             self.assertFalse(generic_flags in tc.get_variable(var))
 
+    def test_optarch_aarch64_heuristic(self):
```

Comment:
@geimer don't we want to test the default behaviour for `AARCH64` as well here?

you can put a dummy module in place for GCC 6.x to check that